testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabUserHelper.java

```diff
@@ -71,6 +71,82 @@
         return configInfos;
     }
 
+    public enum UserType {
+        CLOUD(LabConstants.UserType.CLOUD),
+        B2C(LabConstants.UserType.B2C),
+        FEDERATED(LabConstants.UserType.FEDERATED),
+        GUEST(LabConstants.UserType.GUEST),
+        MSA(LabConstants.UserType.MSA),
+        ON_PREM(LabConstants.UserType.ON_PREM)
+        ;
+        String constant;
+        UserType(String constant) {
+            this.constant = constant;
+        }
+        String getValue() {
+            return constant;
+        }
+
+    }
+
+    public enum SignInAudience {
+        AZURE_AD_AND_PERSONAL_MICROSOFT_ACCOUNT(LabConstants.SignInAudience.AZURE_AD_AND_PERSONAL_MICROSOFT_ACCOUNT),
+        AZURE_AD_MULTIPLE_ORGS(LabConstants.SignInAudience.AZURE_AD_MULTIPLE_ORGS),
+        AZURE_AD_MY_ORG(LabConstants.SignInAudience.AZURE_AD_MY_ORG),
+        ;
+        String constant;
+        SignInAudience(String constant) {
+            this.constant = constant;
+        }
+        String getValue() {
+            return constant;
+        }
+
+    }
+
+    public enum AzureEnvironment {
+        AZURE_B2C_CLOUD(LabConstants.AzureEnvironment.AZURE_B2C_CLOUD),
+        AZURE_CHINA_CLOUD(LabConstants.AzureEnvironment.AZURE_CHINA_CLOUD),
+        AZURE_CLOUD(LabConstants.AzureEnvironment.AZURE_CLOUD),
+        AZURE_GERMANY_CLOUD(LabConstants.AzureEnvironment.AZURE_GERMANY_CLOUD),
+        AZURE_PPE(LabConstants.AzureEnvironment.AZURE_PPE),
+        AZURE_US_GOVERNMENT(LabConstants.AzureEnvironment.AZURE_US_GOVERNMENT)
+        ;
+        String constant;
+        AzureEnvironment(String constant) {
+            this.constant = constant;
+        }
+        String getValue() {
+            return constant;
+        }
+    }
+
+    public enum IsAdminConsented {
+        YES(LabConstants.IsAdminConsented.YES),
+        NO(LabConstants.IsAdminConsented.NO)
+        ;
+        String constant;
+        IsAdminConsented(String constant) {
+            this.constant = constant;
+        }
+        String getValue() {
+            return constant;
+        }
+    }
+
+    public enum PublicClient {
+        YES(LabConstants.PublicClient.YES),
+        NO(LabConstants.PublicClient.NO)
+        ;
+        String constant;
+        PublicClient(String constant) {
+            this.constant = constant;
+        }
+        String getValue() {
+            return constant;
+        }
+    }
+
```

Comment:
Why duplicate all of these as enums? Are the String constants in `LabConstants.java` not sufficient?

Reply:
I'm going to say no, they're not.  Because nothing stops you from putting the wrong string, or nay, any string into the method arguments nothing saves you from transposing the arguments or provides a path from the method to what the potential values are unless you happen to know that there's a class called LabConstants.

Reply:
Makes sense